stor/dx.py

```diff
@@ -824,12 +828,60 @@ def upload(self, to_upload, **kwargs):
                 raise NotFoundError('Source path ({}) does not exist. Please provide '
                                     'a valid source'.format(upload_obj.source))
 
-    def open(self, mode='r', encoding=None):
+    def open(self, mode='r', encoding=None, **kwargs):
         """
-        Opens a OBSFile that can be read or written to and is uploaded to
-        the remote service.
+        Args:
+            mode (str): The mode of the resource. Can be "r" for
+                reading the resource, "w" for writing the
+                resource, and "a" for appending to the open resource.
+            encoding (str): the text encoding to use on read/write, defaults to
+                ``locale.getpreferredencoding(False)`` if not set. We *strongly* encourage you to
+                explicitly set an encoding when reading/writing text (because
+                writers from different computers may store data on DNAnexus in different ways).
+                Python 3 only.
```

Comment:
can you list out what kwargs are allowed to be passed (or reference a section in the docs with the preferred encoding)?

Reply:
So, kwargs are only passed to write_object, which is in turn only passed to upload. Right now, we don't support any kwargs, since we mentioned not exposing the user to all the arguments in dxpy. But, it's easy to support all the parameters in say, `http://autodoc.dnanexus.com/bindings/python/current/dxpy_dxfile.html#dxpy.bindings.dxfile_functions.upload_local_file`, if the users do end up needing such a functionality. All I would need to change is use the passed kwargs in `DXPath.upload` and do `dxpy.upload_local_file(
                            filename=upload_obj.source,
                            project=self.canonical_project,
                            folder=dest_file.parent.resource or '/',
                            parents=True,
                            name=dest_file.name, 
**kawrgs
                        )`